dcpy/metadata/models.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+from pathlib import Path
+from pydantic import BaseModel, conlist
+from typing import Any, Literal
+import yaml
+
+
+class BytesDestination(BaseModel, extra="forbid"):
+    type: Literal["bytes"]
+    id: str
+    datasets: list[str]
+
+
+class SocrataDestination(BaseModel, extra="forbid"):
+    id: str
+    type: Literal["socrata"]
+    four_four: str
+    attachments: list[str] = []
+    datasets: conlist(item_type=str, max_length=1)  # type:ignore
+    omit_columns: list[str]
+    column_details: dict[str, SocrataColumn] = {}
+
+    def destination_column_metadata(self, metadata: Metadata) -> list[SocrataColumn]:
+        soc_cols = []
+        dataset = metadata.dataset_package.get_dataset(self.datasets[0])
+        for col in dataset.get_columns(metadata):
+            if col.name in self.omit_columns:
+                continue
+            overrides = self.column_details.get(col.name, SocrataColumn())
+
+            soc_cols.append(
+                SocrataColumn(
+                    name=col.name,
+                    api_name=overrides.api_name or col.name,
+                    display_name=overrides.display_name or col.display_name,
+                    description=overrides.description or col.description,
+                )
+            )
+        return soc_cols
+
+
+class Column(BaseModel, extra="forbid"):
+    name: str
+    display_name: str
+    description: str
+    data_type: str
+
+    # Optional
+    data_source: str | None = None
+    example: Any | None = None
+    is_nullable: bool = True
+    is_primary_key: bool = False
+    readme_data_type: str | None = None
+    values: list[tuple] = []
```

Comment:
Is mutable default args an issue in pydantic models?

Reply:
Ignore me, it's fine (thank goodness)

Reply:
Thankfully not!